### PR TITLE
TransactionBuilder: shuffle inputs too in Shuffle() method

### PR DESCRIPTION
**Pull request by knocte**
While we are at it, we improve the obsolete message of the Shuffle
method and put the shuffle operations closer together in the
BuildTransaction method.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
No do not do this please, things are working already as it should.

**Comment by knocte** [comment]
> No do not do this please, things are working already as it should.

Can you elaborate? AFAIU the code is much more readable this way, did I break something?

**Comment by NicolasDorier** [comment]
Yes `DoShuffle` != `DoShuffleGroups`. `DoShuffleGroups` does not shuffle inputs by design, because that's why it is called `DoShuffleGroups`

**Comment by knocte** [comment]
My change doesn't make DoShuffleGroups shuffle inputs.

File: NBitcoin/TransactionBuilder.cs [hunk 1]
```diff
@@ -1623,7 +1639,6 @@ public Transaction BuildTransaction(bool sign, SigHash sigHash)
 		public Transaction BuildTransaction(bool sign, SigningOptions signingOptions)
 		{
 			int totalRepass = 5;
-			DoShuffleGroups();
```
Comment (NicolasDorier) [inline NBitcoin/TransactionBuilder.cs:1626]: groups are shuffled before the rest, there is a reason for that, probably why tests break.

Comment (knocte) [reply NBitcoin/TransactionBuilder.cs:1626]: But in my PR DoShuffleGroups() is the first thing that DoShuffle() does.

Comment (NicolasDorier) [reply NBitcoin/TransactionBuilder.cs:1626]: No your code is not equivalent, you are not shuffling group before building the transaction, which make it useless.

Comment (knocte) [reply NBitcoin/TransactionBuilder.cs:1626]: ah, so with "the rest" you don't mean "shuffling the rest" but more stuff, got it; sorry for taking your time

**Pull request closed without merging**